### PR TITLE
ffn.hpp and rnn.hpp: Documentation fix

### PR DESCRIPTION
**Pull request by saksham189**
Documentation fix for EvaluateWithGradient.

### CODE REVIEW

File: src/mlpack/methods/ann/rnn_impl.hpp [hunk 1]
```diff
@@ -304,13 +304,10 @@ EvaluateWithGradient(const arma::mat& /* parameters */,
       responseSeq = seqNum;
     }
 
-    if (!deterministic)
```
Comment (rcurtin) [inline src/mlpack/methods/ann/rnn_impl.hpp]: I agree with the documentation changes, but I'm really not sure about this one.  Why should we remove the `if (!deterministic)` here?

Comment (saksham189) [reply src/mlpack/methods/ann/rnn_impl.hpp]: I think on this https://github.com/mlpack/mlpack/pull/1615/files#diff-0e560b4ae5054e7fe4c029ec9eb0f1a7R275
we set `deterministic` to false.

Comment (saksham189) [reply src/mlpack/methods/ann/rnn_impl.hpp]: `EvaluateWithGradient` is only being used for training otherwise we could call `Evaluate`.

Comment (rcurtin) [reply src/mlpack/methods/ann/rnn_impl.hpp]: Thanks for the clarification; that makes sense to me.

**Comment by ShikharJ** [comment]
@saksham189 Thanks a lot for the contribution.

**Pull request merged**